community/kernel/src/main/java/org/neo4j/kernel/impl/newapi/NodeCursor.java

```diff
@@ -91,7 +94,8 @@ void single( long reference, Read read )
         this.highMark = NO_ID;
         this.read = read;
         this.hasChanges = HasChanges.MAYBE;
-        this.addedNodes = null;
+        this.addedNodes = emptySet();
+        this.single = true;
```

Comment:
I think you can use `this.highMark == NO_ID` to check for `single` without introducing a new field.

Reply:
Thx! was blind or something. Switch to use that method instead.